qt/python/mantidqt/mantidqt/widgets/instrumentview/test/test_instrumentview_manager.py

```diff
@@ -0,0 +1,38 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantidqt package
+#
+#
+import unittest
+
+from mantid.simpleapi import CreateSampleWorkspace
+from mantidqt.widgets.instrumentview.presenter import InstrumentViewManager
+from mantidqt.widgets.instrumentview.api import get_instrumentview
+
+from mantidqt.utils.qt.testing import start_qapplication
+
+
+@start_qapplication
```

Comment:
Can we do this in terms of a presenter test, not a view test?
I'm nervous to add more Qt Application tests as we've just had 2 go flaky in the last development cycle. It's also the reason we created the rule we don't put any logic into the view (which this doesn't) because of testing problems in the paste [(link)](https://developer.mantidproject.org/Standards/GUIStandards.html#cant-test-in-view)

If you changed the presenter around slightly:
```
# From this:
 del InstrumentViewManager.view_dict[ws_name]
# To this:
InstrumentViewManager.view_dict[ws_name] = None
```
Which does the same thing in Python (as `del` is simply a suggestion to trigger GC the same way that setting it to None would be)

We could simply inject a mock in a presenter test:

```
InstrumentViewManager.view_dict = mock.MagicMock()  # Has to be magic mock for [] operator
presenter.remove(...)
self.assertIsNone(InstrumentViewManager.view_dict[ws_name][loc]
```

This would completely remove the need to check via the view at all, would avoid Qt being flaky in the future and if it leaked memory would clearly point to either someone taking another ref, or Python being obstinate releasing memory (which this test wouldn't find either of anyway)

Reply:
The purpose of the test is to verifty that the correct presenter function is called when the view is closed, so I've modified it a bit to make it more specific.